module/VuFind/tests/unit-tests/src/VuFindTest/Sitemap/Plugin/Index/CursorMarkIdFetcherTest.php

```diff
@@ -181,21 +180,29 @@ public function testFetching(): void
         $records2 = new RecordCollection(['nextCursorMark' => 'nextCursor']);
         $expectedIds2 = $this->addRecordsToCollection($records2, $this->countPerPage);
         $service = $this->getMockService();
+        $commandObj = $this->getMockBuilder(\VuFindSearch\Command\AbstractBase::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $commandObj->expects($this->exactly(2))->method('getResult')
+            ->willReturnOnConsecutiveCalls(
+                $this->returnValue($records1),
+                $this->returnValue($records2)
+            );
 
         // Set up all the expected commands...
-        $service->expects($this->exactly(2))->method('invoke')
+        $service->expects($this->exactly(4))->method('invoke')
             ->withConsecutive(
                 [$this->isInstanceOf(GetUniqueKeyCommand::class)],
+                [$this->isInstanceOf(GetIdsCommand::class)],
                 [$this->isInstanceOf(GetUniqueKeyCommand::class)],
+                [$this->isInstanceOf(GetIdsCommand::class)],
             )->willReturnOnConsecutiveCalls(
                 $this->getMockKeyCommand(),
+                $commandObj,
                 $this->getMockKeyCommand(),
+                $commandObj
             );
-        $service->expects($this->exactly(2))->method('legacyInvoke')
-            ->withConsecutive(
-                [$this->callback($this->getIdsExpectation('*'))],
```

Comment:
Have these expectations gotten lost in the refactoring? Do we need to add consecutive expectations to `$commandObj` above to retain this logic?